code/lib/cli/src/automigrate/helpers/getMigrationSummary.ts

```diff
@@ -59,11 +60,15 @@ export function getMigrationSummary({
   installationMetadata?: InstallationMetadata | null;
   logFile: string;
 }) {
+  const automigrateCommand = isPrerelease(versions.storybook)
+    ? 'npx storybook@next automigrate'
+    : 'npx storybook@latest automigrate';
```

Comment:
If someone runs `npx storybook@next upgrade` and after the upgrade someone wants to rerun the automigrations, shouldn't we just tell the users to call `npx storybook automigrate` to run the local version? This makes the most sense to me. Because people might run a specific version of an upgrade like `npx storybook@8.1.0 upgrade` and then the above commands again do not fit, because the user wants to run the `automigrate` command from 8.1.0 and not from `latest` or `next`.

Reply:
I don't think that that's a great idea, we should optimize for the common cases.
Before this PR, the suggested command to run was always with `@next`, this seems important to fix ASAP?

> Because people might run a specific version of an upgrade like npx storybook@8.1.0 upgrade and then the above commands again do not fit

True, but it's not super common to happen?
There's possibly even more edge cases?

The code is also an exact copy of how we already this the exact thing somewhere else already:
https://github.com/storybookjs/storybook/blob/d52adbb107d115c0dd1530038950cd213a315526/code/lib/cli/src/doctor/index.ts#L144-L146

Reply:
If a user runs a command to upgrade Storybook and it fails to upgrade, then if we tell them to run `npx storybook <command>` that will use a version we might not want, right?

Reply:
@yannbf I might be mistaken, but the migration summary isn't shown when the `upgrade` fails. It should only tell the user that the `automigrate` CLI can be re-executed when automigrations are skipped, but the upgrade itself was successful. Otherwise, only running the `automigrate` command when the `upgrade` fails doesn't make much sense. You don't know whether the correct dependencies are already installed or in which phase the upgrade failed.

@ndelangen, Can you elaborate on your concerns about why it isn't a good idea? I am not convinced by the arguments yet. The argument that things traditionally were already done in a specific fashion doesn't tell anything about the quality of an idea or why it is good or bad.

We explicitly worked on the idea to enable people to upgrade to specific versions of Storybook. So, we have already identified this use case as pretty important, and we want to acknowledge it and not treat it as an edge-case.

If `next` is 8.4 and someone upgrades to `8.3` via `npx storybook@8.3.0 upgrade` it is a terrible idea to run automigrations from `8.4.0`. I understand @yannbf argument, that 8.4 might have automigration improvement/fixes, which `8.3.0` doesn't have, on the other hand, the automigration scripts are versioned, and therefore they are coupled with the rest of the Storybook's ecosystem, opening the door for terrible bugs. Second, if Storybook 9 is published, and people upgrade from 7 to 8, should they run `npx storybook@next automigrate` or `npx storybook@latest automigrate`, which both point to a Storybook 9 version? I don't think that's what we want.